wal_e/blobstore/file/calling_format.py

```diff
@@ -0,0 +1,96 @@
+import os
+from datetime import datetime
+
+
+def remove_empty_dirs(path):
+    """ removes empty dirs under a given path """
+    for root, dirs, files in os.walk(path):
+        for d in dirs:
+            dir_path = os.path.join(root, d)
+            if not os.listdir(dir_path):
+                os.removedirs(dir_path)
+
+
+def ensure_dir_exists(path):
+    """ create a directory if required """
+    dir_path = os.path.dirname(path)
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+
+
+def common_dir_path(args, sep='/'):
+    """ return the highest common directory given a list of files """
+    return os.path.commonprefix(args).rpartition(sep)[0]
+
+
+def epoch_to_iso8601(timestamp):
+    return datetime.utcfromtimestamp(timestamp).isoformat()
+
+
+class FileKey:
```

Comment:
For historical reasons, in Python2 use "new-style" classes, which would be: "Filekey(object)" here.